packages/cli/src/lib/JdkInstaller.ts

```diff
@@ -79,12 +85,30 @@ export class JdkInstaller {
     const dstPath = path.resolve(installPath);
     const downloadSrcUrl = DOWNLOAD_JDK_SRC_ROOT + JDK_SRC_ZIP;
     const localSrcZipPath = this.joinPath(dstPath, JDK_SRC_ZIP);
-    await util.downloadFile(downloadSrcUrl, localSrcZipPath);
+
+    const progressBar = new Bar({
+      format: ` >> [${green('{bar}')}] {percentage}% | {value}k of {total}k`,
+    }, Presets.shades_classic);
+
+    this.prompt.printMessage(messages.messageDownloadJdkSrc);
+    progressBar.start(Math.round(JDK_SOURCE_SIZE / 1024), 0);
+    await util.downloadFile(downloadSrcUrl, localSrcZipPath, (current) => {
+      progressBar.update(Math.round(current / 1024));
+    });
+    progressBar.stop();
+
+    this.prompt.printMessage(messages.messageDecompressJdkSrc);
```

Comment:
I feel you could extract this and the chunk you added below into some method.

Potentially even something static that could also be used in the Android downloader.

Reply:
+1

Reply:
+1

Reply:
done.